ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/bonsai/trielog/TrieLogPruner.java

```diff
@@ -50,11 +53,13 @@ public class TrieLogPruner {
   public TrieLogPruner(
       final BonsaiWorldStateKeyValueStorage rootWorldStateStorage,
       final Blockchain blockchain,
+      final Consumer<Runnable> executeAsync,
```

Comment:
Might have been more obvious to pass EthScheduler in here but that would mean making the `core` module depend on the `eth` module which is circular.

I do like the way this is decoupled from the EthScheduler implementation anyway, even if the type is a little unobvious.

Reply:
The `Executor` would be clearer than using a `Consumer<Runnable>` type

Reply:
As discussed, while it happens to be the same type, the implementation I'm using does not actually represent an Executor implementation.